src/Symfony/Bridge/PhpUnit/Tests/ProcessIsolationTest.php

```diff
@@ -10,6 +10,12 @@
  * @group legacy
  *
  * @runTestsInSeparateProcesses
+ * @preserveGlobalState disabled
+ *
+ * Note that for the deprecation handler to work in a separate process we need to disable the preservation of global
+ * state. This is because composer's autoloader stores which files have been autoloaded in the global
+ * '__composer_autoload_files'. If this is preserved then bootstrap.php will not run again meaning that deprecations
+ * won't be collected.
```

Comment:
Wouldn't it be enough to reset the autoloaded state then?

Reply:
For the use case of deprecating an interface (and other scenarios), you'd add `@trigger_error()` after the `namespace` declaration. That means the error will be triggered on discovery and then never repeated, for a false positive.

Disabling `@preserveGlobalState` here implies to me that for some process-isolated tests to reflect all deprecations, they should have this same annotation, so a documentation update would be needed, and/or the test listener needs to set this value on isolated tests.

Reply:
@xabbuh I'm not sure when we'd get a chance to do that? We need bootstrap.php to be loaded before the test is run in isolation.

Reply:
Worked out something about the failing tests on travis for 7.2 - it's because of something in src/Symfony/Bridge/PhpUnit/bin/simple-phpunit - i've downloaded the travis docker image and replayed the build script. Running phpunit via that script it fails. Running vanilla phpunit it passes.